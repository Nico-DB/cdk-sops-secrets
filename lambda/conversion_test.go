package main

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func Test_FromYAML(t *testing.T) {
	simpleStruct, err := fromYAML(ReadFile("../test-secrets/yaml/sopsfile.yml"))
	check(err)
	snaps.MatchSnapshot(t, simpleStruct)
	complexStruct, err := fromYAML(ReadFile("../test-secrets/yaml/sopsfile-complex.yml"))
	check(err)
	snaps.MatchSnapshot(t, complexStruct)
}

func Test_toJSON(t *testing.T) {
	simpleJSON, err := toJSON(UnmarshalAny(ReadFile("../test-secrets/json/sopsfile.json")))
	check(err)
	snaps.MatchSnapshot(t, string(simpleJSON))
	complexJSON, err := toJSON(UnmarshalAny(ReadFile("../test-secrets/json/sopsfile-complex.json")))
	check(err)
	snaps.MatchSnapshot(t, string(complexJSON))
}
