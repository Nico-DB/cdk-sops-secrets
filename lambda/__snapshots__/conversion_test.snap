
[Test_FromYAML - 1]
map[string]interface {}{
    "key1": "value1",
    "key2": int(12345),
    "key3": bool(false),
}
---

[Test_FromYAML - 2]
map[string]interface {}{
    "some": map[string]interface {}{
        "deep": map[string]interface {}{
            "nested": map[string]interface {}{
                "arrays": []interface {}{
                    "with",
                    "several",
                    map[string]interface {}{
                        "values": map[string]interface {}{
                            "and": "objects",
                        },
                    },
                },
                "object": "structure",
            },
        },
        "notsodeep": "struct",
    },
}
---

[Test_toJSON - 1]
{"key1":"value1","key2":12345,"key3":false}
---

[Test_toJSON - 2]
{"some":{"deep":{"nested":{"arrays":["with","several",{"values":{"and":"objects"}}],"object":"structure"}},"notsodeep":"struct"}}
---
